.github/workflows/contrib-openai.yml

```diff
@@ -13,7 +13,7 @@ on:
       - 'setup.py'
 
 jobs:
-  RetrieveChatTest:
+  OpenAI4ContribTests:
```

Comment:
One problem with this approach is that the previous installed dependencies will remain in the later steps. That makes the test environment not clean for later steps. Is it possible to reset the environment for later steps?

Reply:
I haven't found any good way to clear up the environment. A possible solution is to use virtual env, but different os needs different syntax, which makes it complex. The other way is just putting them in different jobs.

I guess there are two problems to consider here:
1. Is it ok to NOT having a clean env? Will there be a case that user does `pip install autogen[retrievechat, mathchat]`, and they expect things just work?
2. I suggested that separating the jobs would be tedious to look in github actions, which is why I change this in the first place. Do you think it is indeed a issue, or we can bear with it?

Reply:
I think it's pretty important for the automatic jobs to test different contributed agents in different environments, each with just its own dependencies, whether this is done through separate virtual envs, different jobs, or some other mechanism. Wish I had more experience in setting up such tests.

Reply:
I just realize that `coverage run -a -m pytest test/test_retrieve_utils.py test/agentchat/contrib` will require dependencies from different contributed agents